### PR TITLE
WIP: Test and docs

### PR DESCRIPTION
**Pull request by mfalt**
* Includes the PR https://github.com/JuliaControl/ControlSystems.jl/pull/212
* Updates tests to require plots to not fail and docs-tests to pass.
* ~This means that ControlExamplePlots.jl is needed to run tests now~
* Have copy of code to generate test plots locally in test
* Updated so docs-tests are run on all examples in documentation, including plots 
* Removed duplicate code for generating documentation plots
* Temporary fix to `jldoctest` to be able to hide lines using # hide
* Other smaller fixes

Builds on top of https://github.com/JuliaControl/ControlSystems.jl/pull/256 to avoid rewriting doctests again

### CODE REVIEW

File: .github/workflows/CompatHelper-test.yml [hunk 1]
```diff
@@ -1,8 +1,8 @@
 name: CompatHelper
 
 on:
-  schedule:
```
Comment (mfalt) [inline .github/workflows/CompatHelper-test.yml:4]: This is a mistake from the other PR right? @baggepinnen

Comment (mfalt) [reply .github/workflows/CompatHelper-test.yml:4]: Both the change in the file and the name change

**Comment by mfalt** [comment]
@JuliaControlBot test-plots

**Comment by mfalt** [comment]
Some interesting data from the tests on Julia 1.4:
Timings:
* Total: 16-19 min
* Pkg.test: 15min (several minutes setting up and pre-compiling before tests start)
 
Time |   Test
| ---- | ---- |
| 10min | All tests |
| 14s |  test_statespace |
| 13s  | test_transferfunction |
| 118s  | test_delayed_systems  (mostly compilation) (90s on 1.0) |
| 15s  | test_zpk |
| 3s  | test_promotion |
| 115s  | test_connections  (only 43s on 1.0) |
| 2s  | test_discrete |
| 13s  | test_conversion |
| 14s  | test_complex |
| 29s  | test_linalg  (22s on 1.0) |
| 7s  | test_simplification |
| 7s  | test_freqresp |
| 29s  | test_timeresp |
| 8s  | test_analysis |
| 16s  | test_matrix_comps |
| 4s  | test_lqg |
| 7s  | test_synthesis |
| 0s  | test_partitioned_statespace |
| 0s  | test_demo_systems |
| 168s  | test_plots |
| 18s  | test_plots_in_docs  (i.e testing that we can generate plots needed for docs) |
| 18s  | test_doctests |

After success: 200s to install deps and build docs again

File: docs/src/examples/example.md [hunk 2]
```diff
@@ -92,16 +106,17 @@ P  = tf(B,A)
...
 # output
 
-TransferFunction:
-      1.0
-----------------
-s^2 + 0.4s + 1.0
+TransferFunction{Continuous,ControlSystems.SisoRational{Float64}}
+         1.0
+---------------------
+1.0*s^2 + 0.4*s + 1.0
```
Comment (baggepinnen) [inline docs/src/examples/example.md:112]: How come this has changed? The old output seems much nicer?

File: docs/make.jl [hunk 3]
```diff
@@ -1,20 +1,44 @@
...
+Markdown = Documenter.Documents.Markdown
+function Documenter.Documents.doctest_replace!(block::Markdown.Code)
+    startswith(block.language, "jldoctest") || return false
+    # suppress output for `#output`-style doctests with `output=false` kwarg
+    if occursin(r"^# output$"m, block.code) && occursin(r";.*output\h*=\h*false", block.language)
+        input = first(split(block.code, "# output\n", limit = 2))
+        block.code = rstrip(input)
+    end
+    # Remove # hide lines
+    block.code = Documenter.Expanders.droplines(block.code)
```
[the remark this answers is not in the document]
Comment (mfalt) [reply docs/make.jl:23]: It doesn't have it for `jldoctest`, and the other environments doesn't run when doing doctest. So either we have to build the full docs as part of the test, or we can not test the code in the docs.

File: docs/src/examples/example.md (hunk 2 above)
Comment (mfalt) [reply docs/src/examples/example.md:112]: I think it was a change to Polynimials, that we can not override in a simple way. It happened a while ago, this is just updated to reflect the reality.

File: docs/make.jl (hunk 3 above)
[the remark this answers is not in the document]
Comment (baggepinnen) [reply docs/make.jl:23]: Ah I see :)

File: docs/mkdocs.yml [hunk 4]
```diff
@@ -21,11 +21,11 @@ docs_dir: 'build'
 
 pages:
   - Home: 'index.md'
-  - Examples:
-    - Design: 'examples/example.md'
   - Guide:
```
[the remark this answers is not in the document]
Comment (mfalt) [reply docs/mkdocs.yml]: Only if you manually run mkdocs i think, removing it to keep it simple.

**Comment by baggepinnen** [comment]
This PR ready for merge? @mfalt

File: docs/src/examples/example.md (hunk 2 above)
Comment (olof3) [reply docs/src/examples/example.md:112]: I created an [issue](https://github.com/JuliaMath/Polynomials.jl/issues/275) in `Polynomials.jl` for this and the sometimes excessive precision in the coefficients.

**Comment by mfalt** [comment]
Close by https://github.com/JuliaControl/ControlSystems.jl/pull/350

**Pull request closed without merging**